### ISSUE TITLE
Web - IOU - Unable to delete replaced receipt on manual request

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#40552]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.63-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
1. Go to any chat> Create manual request
2. Go to detail IOU page> Add receipt
3. Go to 3-dots menu and replace the receipt
4. Go to 3-dots menu and select Delete

## Expected Result:
Receipt should be deleted

## Actual Result:
Unable to delete replaced receipt on manual request when there is one request

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/94973ab8-d886-43cf-8701-d8246593a9e6

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kavimuru** [comment]
This bug might be related to #wave-collect - Release 1

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Unable to delete replaced receipt on manual request when there is one request

### What is the root cause of that problem?
The back-end is returning `405 Cannot detach receipt from unowned transaction`, and we're simply restoring the transaction [here](https://github.com/Expensify/App/blob/f28d401b33649e25512cdc88efd4d7d0d944ab5f/src/libs/actions/IOU.ts#L5757) without showing any error message to the user.

So from the user POV, the receipt just restores itself and the feature looks like it doesn't work.

### What changes do you think we should make in order to solve the problem?

1. In here https://github.com/Expensify/App/blob/f28d401b33649e25512cdc88efd4d7d0d944ab5f/src/libs/actions/IOU.ts#L5757, we should set `errors` on the transaction if the receipt deletion fails.
2. Fix the back-end not to error in this case, because it's a valid case

### What alternative solutions did you explore? (Optional)
NA

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Unable to delete replaced receipt on manual request when there is one request

### What is the root cause of that problem?
The back-end is returning `405 Cannot detach receipt from unowned transaction`, and we're simply restoring the transaction [here](https://github.com/Expensify/App/blob/f28d401b33649e25512cdc88efd4d7d0d944ab5f/src/libs/actions/IOU.ts#L5757) without showing any error message to the user.

So from the user POV, the receipt just restores itself and the feature looks like it doesn't work.

### What changes do you think we should make in order to solve the problem?

1. In here https://github.com/Expensify/App/blob/f28d401b33649e25512cdc88efd4d7d0d944ab5f/src/libs/actions/IOU.ts#L5757, we should set `errors` on the transaction if the receipt deletion fails. We can add a new error message key like `Unexpected error deleting this receipt. Please try again later.`

```
value: {
    ...transaction,
    errors: ErrorUtils.getMicroSecondOnyxError('iou.error.receiptDeleteFailureError')
},
```

We already have it for [the replaceReceipt case](https://github.com/Expensify/App/blob/f28d401b33649e25512cdc88efd4d7d0d944ab5f/src/libs/actions/IOU.ts#L5792), and when requesting money, just missing it for the delete receipt case.

We can also remove the error in optimistic/success data.

2. Fix the back-end not to error in this case, because it's a valid case.

### What alternative solutions did you explore? (Optional)
NA

**Comment by allroundexperts** [comment]
@nkdengineer's [proposal](https://github.com/Expensify/App/issues/40552#issuecomment-2071998200) looks good to me. In this case, we'd need someone on the backend to fix the error. Regardless of that, as mentioned in the proposal, we should show the error and that would require a fix from the frontend.

🎀 👀 🎀 C+ reviewed

**Comment by MariaHCD** [comment]
I agree that we need to show an error message if deleting a receipt on a request does not succeed 👍🏼

**Comment by MariaHCD** [comment]
Looking into the backend error and leaving some notes for myself.

We're throwing the error [here](https://github.com/Expensify/Auth/blob/da767cc9b950abdcb9f17db17e7735fe11a009c2/auth/lib/Receipt.cpp#L84-L88)

<details>

<summary> Logs from testing locally </summary>

```
2024-04-29T07:44:01.472184+00:00 expensidev2004 bedrock: USMjjb test@iouaccount.com (Transaction.cpp:46) verifyWrite [sync] [info] Transaction::verifyWrite accountID=9684 transactionIDList=9166041848027912271 maxReportState=0 isValidOwner=1 areOnValidReports=0

2024-04-29T07:44:01.472366+00:00 expensidev2004 bedrock: USMjjb test@iouaccount.com (Receipt.cpp:65) attach [sync] [info] [ReceiptAttach] write/own accountID=9684 receiptID=0 transactionID=9166041848027912271 verifyWrite=0 managerOrAdminAttach=1 scraperMergeAttach=0 isAttachedToPartialTransaction=0

```
</details>

I believe it's because of [this condition](https://github.com/Expensify/Auth/blob/03dc9b461c1812173860a03e53df3400ae6956d6/auth/command/AttachReceipt.cpp#L76). Spinning up a PR.

**Comment by MariaHCD** [comment]
Still working on the backend PR -  https://github.com/Expensify/Auth/pull/10712

**Comment by MariaHCD** [comment]
@nkdengineer I believe you can proceed with the frontend PR while the backend piece is being worked on.

**Comment by nkdengineer** [comment]
@allroundexperts The PR is [here](https://github.com/Expensify/App/pull/41553).

**Comment by MariaHCD** [comment]
The backend PR has been merged and will be deployed later today: https://github.com/Expensify/Auth/pull/10712

**Comment by anmurali** [comment]
@nkdengineer is paid. @allroundexperts can you pls complete the C+ checklist?

**Comment by allroundexperts** [comment]
## Checklist

1. This was something that was never implemented. As such, there isn't any PR to pinpoint to.
2. N/A
3. N/A
4. A regression test would be helpful. 

## Regression test

Pre-Req: Make sure that `Simulate failing network requests` option is turned on in the `Troubleshooting` section.

1. Go to any chat -> Create manual request
2. Go to detail IOU page -> Add receipt
3. Go to 3-dots menu and replace the receipt
4. Go to 3-dots menu and select Delete
5. Wait for the request to fail

Verify that an error shows up under the receipt image.

Do we 👍 or 👎 ?

**Comment by JmillsExpensify** [comment]
$250 approved for @allroundexperts
